lib/ts/recipe/emailverification/recipeImplementation.ts

```diff
@@ -0,0 +1,184 @@
+import Querier from "../../querier";
+import { NormalisedAppInfo } from "../../types";
+import { getQueryParams } from "../../utils";
+import { APIGeneralError, RecipeFunctionOptions } from "../recipeModule/types";
+import { NormalisedInputType, RecipeInterface } from "./types";
+
+const EMAIL_VERIFY_PATH = "/user/email/verify";
+const SEND_VERIFY_EMAIL_PATH = "/user/email/verify/token";
+
+export default function getRecipeImplementation(recipeId: string, appInfo: NormalisedAppInfo): RecipeInterface {
+    const querier = new Querier(recipeId, appInfo);
+    return {
+        verifyEmail: async function ({
+            token,
+            config,
+            options,
+        }: {
+            token?: string;
+            config: NormalisedInputType;
+            options?: RecipeFunctionOptions;
+        }): Promise<{
+            status: "EMAIL_VERIFICATION_INVALID_TOKEN_ERROR" | "OK" | "CUSTOM_RESPONSE";
+            fetchResponse?: Response;
+        }> {
+            token = token === undefined ? getQueryParams("token") : token;
+
+            if (token === undefined) {
+                return {
+                    status: "EMAIL_VERIFICATION_INVALID_TOKEN_ERROR",
+                };
+            }
+
+            const { json, fetchResponse } = await querier.post<
+                | {
+                      status: "OK" | "EMAIL_VERIFICATION_INVALID_TOKEN_ERROR";
+                  }
+                | APIGeneralError
+            >(
+                EMAIL_VERIFY_PATH,
+                {
+                    body: JSON.stringify({
+                        method: "token",
+                        token,
+                    }),
+                },
+                async (context) => {
+                    let postRecipeHookContext = await config.preAPIHook({
+                        ...context,
+                        action: "VERIFY_EMAIL",
+                    });
+
+                    if (options === undefined || options.preAPIHook === undefined) {
+                        return postRecipeHookContext;
+                    }
+
+                    return options.preAPIHook({
+                        url: postRecipeHookContext.url,
+                        requestInit: postRecipeHookContext.requestInit,
+                    });
+                }
+            );
+
+            if (json.status === "GENERAL_ERROR") {
+                throw new Error(json.message);
+            }
+
+            if (json.status !== "OK" && json.status !== "EMAIL_VERIFICATION_INVALID_TOKEN_ERROR") {
+                return {
+                    status: "CUSTOM_RESPONSE",
+                    fetchResponse,
+                };
+            }
```

Comment:
Change code structure to:
```
if (json.status === "OK") {..} else if (json.status === "EMAIL_VERIFICATION_INVALID_TOKEN_ERROR) {...} else {return custom response}
```

Reply:
We no longer need "CUSTOM_RESPONSE", have refactored the code accordingly